.github/workflows/sycl_containers.yaml

```diff
@@ -9,6 +9,9 @@ on:
       - sycl
     paths:
       - 'devops/containers/**'
+  pull_request:
+    paths:
+      - 'devops/containers/**'
 
 jobs:
   base_image_ubuntu2004:
```

Comment:
All three jobs are the same - the only difference is the container name. Can we avoid duplication by applying re-usable workflows or matrix strategy here?

Reply:
This is probably as far as I can go simplifying this file. Matrix builds do not support dependencies, and re-usable workflows would still require me to pass input parameters, which would be identical to what I did with actions.

Reply:
I was aiming to avoid duplication of "push: ${{ github.event_name != 'pull_request' }}" condition.